awscrt/s3.py

```diff
@@ -586,3 +586,18 @@ def is_optimized_for_system():
             for the current system.
     """
     return _awscrt.s3_is_crt_s3_optimized_for_system()
+
+
+def get_recommended_max_throughput_gbps():
+    """
+        Returns:
+        Recommended max throughput in Gbps for the CRT S3 client based on detected system configuration as a PyFloat.
+        If the best throughput configuration is unknown, returns 5Gbps.
+    """
+    # Currently the CRT returns 0 if it was unable to make a good guess on configuration. Pre-known configs, have this value set.
+    # Eventually, the CRT will make a full calculation based on NIC and CPU configuration, but until then handle 0.
+    max_value = _awscrt.s3_get_recommended_max_throughput_gbps()
+    if abs(max_value) > 0:
+        return max_value
+    else:
+        return 5.0
```

Comment:
highly debatable: should we use this new functionality in the S3Client constructor if the user doesn't pass throughput_target_gbps?

or should we use this down in C, instead of the current behavior of defaulting to 10?

Reply:
if not, let's at least add to the S3Client constructor's docs, to mention this function

Reply:
Discussed offline, I have updated it to return None so that we can keep the default value behavior in one place